packages/aws-cdk/bin/cdk.ts

```diff
@@ -284,6 +284,24 @@ async function initCommandLine() {
                 throw e;
             }
         }));
+
+        /**
+         * De-duplicates a list of environments, such that a given account and region is only represented exactly once
+         * in the result.
+         *
+         * @param envs the possibly full-of-duplicates list of environments.
+         *
+         * @return a de-duplicated list of environments.
+         */
+        function distinct(envs: cxapi.Environment[]): cxapi.Environment[] {
+            const unique: { [id: string]: cxapi.Environment } = {};
+            for (const env of envs) {
+                const id = `${env.account || 'default'}/${env.region || 'default'}`;
```

Comment:
Can we somehow utilize the existing concept of environment name (which I believe is rendered in the same way)?